test/test_playbooks/activation_key.yml

```diff
@@ -16,6 +16,9 @@
   - include: tasks/repository.yml
     vars:
       repository_state: present
+  - include: tasks/host_collection.yml
+    vars:
+      host_collection_state: present
```

Comment:
Can you please add some tests here?
immediately before deleting the activation_key, can you check that including the task with `activation_key_host_collections: omit` does not change anything (`expected_change`), while `activation_key_host_collections: []` deletes the associations (`expected_change: true`)?
Honestly, i am not quite sure, this functionality is already in the module, but i think, we want it.

Reply:
Absolutely. Adding this:

```yaml
  - include: tasks/activation_key.yml
    vars:
      activation_key_host_collections: omit
      expected_change: false
  - include: tasks/activation_key.yml
    vars:
      activation_key_host_collections: []
      expected_change: true
```

Reply:
Sorry, not working.

> "No host collection found for omit"

Reply:
Probably because it's hardcoded to be a list in the playbook. I'll see if I can fix it.

Reply:
Try with `...lection: "{{ omit }}"`

Reply:
I think you were right. The `activation_key` module reports changed when host collection is omitted.

Reply:
```
TASK [debug] *******************************************************************
ok: [tests] => {
    "activation_key_host_collections": "__omit_place_holder__5f9bd43780e26a3d6e85302429661028a3df72b3"
}

TASK [Create/update katello activation key] ************************************
changed: [tests] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "auto_attach": false,
            "content_overrides": [
                {
                    "label": "Test_Organization_Test_Product_Test_Repository",
                    "override": "disabled"
                }
            ],
            "content_view": "Default Organization View",
            "host_collections": null,
            "lifecycle_environment": "Library",
            "name": "Test Activation Key",
            "new_name": null,
            "organization": "Test Organization",
            "state": "present",
            "subscriptions": [
                {
                    "name": "Test Product"
                }
            ]
        }
    }
}

TASK [fail] ********************************************************************
task path: /Users/chbr/repos/work/github/foreman-ansible-modules/test/test_playbooks/tasks/activation_key.yml:33
fatal: [tests]: FAILED! => {
    "changed": false,
    "msg": "Creating/updating activation key failed! (expected_change: False)"
}
```

Reply:
Ok I think I've managed to fix it. I removed it as a default and then explicitly added it into the test playbook for some of the tasks. I believe the playbook properly tests all cases now:
- Creating key
- Updating key without host collections
- Updating key with new (empty) host collections
- Deleting key